device_doctor/test/src/ios_device_test.dart

```diff
@@ -46,154 +46,154 @@ void main() {
     });
 
     test('checkDevices with device', () async {
+      process = FakeProcess(0, out: <List<int>>[]);
       when(processManager.start(any, workingDirectory: anyNamed('workingDirectory')))
           .thenAnswer((_) => Future.value(process));
-      process = FakeProcess(0, out: null);
       deviceDiscovery.outputs = <dynamic>['abcdefg'];
       Map<String, List<HealthCheckResult>> results = await deviceDiscovery.checkDevices(processManager: processManager);
       expect(results.keys.length, equals(1));
       expect(results.keys.toList()[0], 'ios-device-abcdefg');
-      expect(results['ios-device-abcdefg'].length, 7);
-      expect(results['ios-device-abcdefg'][0].name, kDeviceAccessCheckKey);
-      expect(results['ios-device-abcdefg'][0].succeeded, true);
-      expect(results['ios-device-abcdefg'][1].name, kKeychainUnlockCheckKey);
-      expect(results['ios-device-abcdefg'][1].succeeded, true);
-      expect(results['ios-device-abcdefg'][2].name, kCertCheckKey);
-      expect(results['ios-device-abcdefg'][2].succeeded, false);
-      expect(results['ios-device-abcdefg'][3].name, kDevicePairCheckKey);
-      expect(results['ios-device-abcdefg'][3].succeeded, false);
-      expect(results['ios-device-abcdefg'][4].name, kUserAutoLoginCheckKey);
-      expect(results['ios-device-abcdefg'][4].succeeded, false);
-      expect(results['ios-device-abcdefg'][5].name, kDeviceProvisioningProfileCheckKey);
-      expect(results['ios-device-abcdefg'][5].succeeded, false);
-      expect(results['ios-device-abcdefg'][6].name, kBatteryLevelCheckKey);
-      expect(results['ios-device-abcdefg'][6].succeeded, false);
+      expect(results['ios-device-abcdefg']?.length, 7);
```

Comment:
nit: You could do `expect(results['ios-device-abcdefg'], isNotNull);` first, then you won't need the extra `?` checks.

Also, doing a single `!` will assert it's not null for the rest of the code.

Reply:
Updated.